### PR TITLE
docs(deployment): improve Docker reverse proxy documentation

### PR DESCRIPTION
**Pull request by camiloforero**
I've been working on deploying an httos cubejs instance using `docker-compose` over the past few days. Although it was good enough for me to finally make it, as an absolute Docker Compose beginner I ran into some issues that might have been obvious to a more experienced dev, but they weren't for me. Although it was a cool learning experience, I think updating the documentation to be more accurate and work out of the box will make things easier for those ahead.

The changes are:
- Specify that the `cube.conf` file goes inside a `nginx` directory, otherwise nothing will be mounted in the corresponding
- Don't mount the `nginx` directly on top of `etc/nginx`, do it on `etc/nginx/conf.d` instead, otherwise it will overwrite existing nginx configuration, and the webserver will fail to start.
- In the nginx `proxy_pass` configuration, change the `localhost` reference to a `cube` reference; in Docker Compose it appears that each service has it's own separate `localhost`, and that configuration in Linux would cause nginx to fail with a 502 error. Referencing the `cube` service instead works as expected.
- Add some extra helpful lines, and a link to a good resource I found and followed to set up SSL with LetsEncrypt.
- Lastly, I changed the `webserver` service name to `nginx`. It's a small change and shouldn't be too important one way or another, but in this case the linked resource assumes the nginx service is named `nginx`, so why not

**Check List**
- [ ] Tests has been run in packages where changes made if available
- [ ] Linter has been run for changed code
- [ ] Tests for the changes have been added if not covered yet
- [ ] Docs have been added / updated if required

**Issue Reference this PR resolves**

[For example #12]

**Description of Changes Made (if issue reference is not provided)**

[Description goes here]

### CODE REVIEW

File: docs/content/Deployment/Platform-Docker.mdx [hunk 1]
```diff
@@ -173,15 +173,19 @@ Then we'll add a new service to our Docker Compose stack:
...
       - 443:443
     volumes:
-      - ./nginx:/etc/nginx
+      - ./nginx:/etc/nginx/conf.d
       - ./ssl:/etc/ssl/private
 ```
 
+Don't forget to create a `ssl` directory with the `cert.pem` and `key.pem` files inside so the Nginx service can find them.
+
+For automatically provisioning SSL certificates with LetsEncrypt, [this blog post](https://pentacent.medium.com/nginx-and-lets-encrypt-with-docker-in-less-than-5-minutes-b4b8a60d3a71) may be useful
```
Comment (hassankhan) [inline docs/content/Deployment/Platform-Docker.mdx:187]:
```suggestion
For automatically provisioning SSL certificates with LetsEncrypt, [this blog post](https://pentacent.medium.com/nginx-and-lets-encrypt-with-docker-in-less-than-5-minutes-b4b8a60d3a71) may be useful.
```

**Review by hassankhan — CHANGES_REQUESTED** [review]
Thanks @camiloforero for making this contribution 💯 

Just found a few minor nits, if you wouldn't mind having a look 😀

**Review by hassankhan — APPROVED** [review]

**Comment by camiloforero** [comment]
@hassankhan thank you! I've merged your suggestions

**Comment by hassankhan** [comment]
@camiloforero Thanks again for your contribution 🥇

**Pull request merged**